src/WhatToCook.Application/Services/MealPlanningService.cs

```diff
@@ -38,27 +45,39 @@ public async Task<PlanOfMeals> Create(PlanOfMealRequest planOfMealRequest)
     public async Task<PlanOfMeals> Update(UpdatePlanOfMealRequest planOfMealRequest)
     {
         var mealPlanToUpdate = await _mealPlanningRepository.GetMealPlanByName(planOfMealRequest.Name);
-        var RecipeForMealPlanUpdate = planOfMealRequest.Recipes.Select(x => x.Name);
 
-        var recipes = _recipesRepository.GetByNames(RecipeForMealPlanUpdate);
         if (mealPlanToUpdate == null)
         {
-            _logger.LogError($"Attempted to update a non-existent meal plan: {planOfMealRequest.Name}");
+            _logger.LogError("Attempted to update a non-existent meal plan {planOfMealName}", planOfMealRequest.Name);
             throw new NotFoundException(nameof(mealPlanToUpdate));
         }
-        //check if all recipes in the request exist in the database
-        var existingRecipes = recipes.Select(r => r.Name).ToList();
-        if (!existingRecipes.OrderBy(n => n).SequenceEqual(RecipeForMealPlanUpdate.OrderBy(n => n)))
+
+        var RecipeForMealPlanUpdate = planOfMealRequest.Recipes.SelectMany(x => x.RecipeIds);
+        var recipes = _recipesRepository.GetRecipesByIdForMealPlan(RecipeForMealPlanUpdate);
+
+        var existingRecipes = recipes.Select(r => r.Id).ToList();
+
+        var missingRecipes = RecipeForMealPlanUpdate.Except(existingRecipes).ToList();
+        if (missingRecipes.Any())
         {
-            _logger.LogError($"Some recipes do not exist in the database");
-            throw new NotFoundException($"Some recipes do not exist in the database");
+            var missingIdsMessage = "Some recipes do not exist in the database. Missing IDs: " + string.Join(", ", missingRecipes);
+
+            _logger.LogError("Some recipes do not exist in the database. Missing IDs: {MissingRecipeIds}", missingIdsMessage);
```

Comment:
Parameter should be changed to string.Join(", ", missingRecipes)

Reply:
fixed, will be in the next commit